### PR TITLE
Add cancel subscription button

### PR DESCRIPTION
**Pull request by jesusantguerrero**

### CODE REVIEW

**Comment by jesusantguerrero** [comment]
Note: ask for cancel subscription endpoint

File: src/components/companies/subscriptions/details.vue [hunk 1]
```diff
@@ -24,13 +24,18 @@
                     <span>Credit card ending in <strong>{{ paymentData.payment_ending_numbers }}</strong></span>
                     <a href="#" @click.prevent="updatePaymentMethod()">Update</a>
                 </div>
+                <div class=" mt-2">
+                    <button class="btn btn-outline-danger" @click="cancelSubscription">
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]:
```suggestion
                    <button class="btn btn-outline-danger" @click="cancelSubscription()">
```

File: src/components/companies/subscriptions/details.vue [hunk 2]
```diff
@@ -114,6 +122,29 @@ export default {
                 name: "plans-modal",
                 scrollable: true
             });
+        },
+
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue:126]:
```suggestion
```

File: src/components/companies/subscriptions/details.vue [hunk 3]
```diff
@@ -114,6 +122,29 @@ export default {
...
+        cancelSubscription() {
+            if (confirm("Are you sure you want to cancel your subscription?")) {
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]: Please, no confirm. We have modal components with proper elegant design.

**Review by SparoHawk — CHANGES_REQUESTED** [review]

File: src/components/companies/subscriptions/details.vue (hunk 3 above)
Comment (jesusantguerrero) [reply src/components/companies/subscriptions/details.vue]: @SparoHawk  done, I am using the modal now.

File: src/components/companies/subscriptions/details.vue [hunk 4]
```diff
@@ -24,13 +24,18 @@
                     <span>Credit card ending in <strong>{{ paymentData.payment_ending_numbers }}</strong></span>
                     <a href="#" @click.prevent="updatePaymentMethod()">Update</a>
                 </div>
+                <div class=" mt-2">
+                    <button class="btn btn-outline-danger" @click.prevent="confirmCancelSubscription">
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]:
```suggestion
                    <button type="button" class="btn btn-outline-danger" @click="confirmCancelSubscription">
```

File: src/components/companies/subscriptions/details.vue [hunk 5]
```diff
@@ -114,6 +122,50 @@ export default {
...
+        confirmCancelSubscription() {
+            this.$modal.show("basic-modal", {
+                title: "Are you sure?",
+                message: "Are you sure you want to cancel your subscription?",
+                buttons: [
+                    {
+                        class: "btn-success",
+                        title: "Yes",
+                        handler: () => {
+                            this.cancelSubscription()
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]:
```suggestion
                            this.cancelSubscription();
```

File: src/components/companies/subscriptions/details.vue [hunk 6]
```diff
@@ -114,6 +122,50 @@ export default {
...
+                        }
+                    }, {
+                        class: "btn-danger",
+                        title: "No",
+                        handler: () => {
+                            this.$modal.hide("basic-modal")
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]:
```suggestion
                            this.$modal.hide("basic-modal");
```

File: src/components/companies/subscriptions/details.vue [hunk 7]
```diff
@@ -114,6 +122,50 @@ export default {
...
+                });
+                this.updateSubscriptionData({});
+            }).catch((error) => {
+                this.$notify({
+                    title: "Error",
+                    text: error.response.data.errors.message,
+                    type: "error"
+                });
+            }).finally(() => {
+                this.$modal.hide("basic-modal")
```
Comment (SparoHawk) [inline src/components/companies/subscriptions/details.vue]:
```suggestion
                this.$modal.hide("basic-modal");
```

**Review by SparoHawk — CHANGES_REQUESTED** [review]

**Review by SparoHawk — APPROVED** [review]

**Pull request merged**